core/common/src/main/java/alluxio/collections/FieldIndex.java

```diff
@@ -22,13 +22,14 @@
  *
  * @param <T> type of objects in this {@link IndexedSet}
  */
-interface FieldIndex<T> extends Iterable<T> {
+public interface FieldIndex<T> extends Iterable<T> {
   /**
    * Adds an object o to the index.
    *
    * @param o the object to add to the index
+   * @return true if success, otherwise false
```

Comment:
Could you be a bit more explicit in what success means? For example Java's `Set.add`: https://docs.oracle.com/javase/7/docs/api/java/util/Set.html#add(E)